_rules/image-not-in-acc-tree-is-decorative-e88epe.md

```diff
@@ -0,0 +1,271 @@
+```
+
+#### Passed Example 3
+
+This decorative `img` element is ignored by assistive technologies because it has an [explicit semantic role][] of `none`.
+
+```html
+Happy new year!<br />
+<img src="/test-assets/shared/fireworks.jpg" role="none" alt="ignore me" />
+```
+
+#### Passed Example 4
+
+This decorative `svg` element is ignored by assistive technologies because it has no attribute that would give it an [accessible name][].
+
+```html
+Happy new year!<br />
+<svg height="200" xmlns="http://www.w3.org/2000/svg">
+	<polygon points="100,10 40,180 190,60 10,60 160,180" fill="yellow" />
+</svg>
+```
+
+#### Passed Example 5
+
+This decorative `canvas` element is ignored by assistive technologies because it has no attribute that would give it an [accessible name][].
+
+```html
+Happy new year!<br />
+<canvas id="newyear" width="200" height="200"></canvas>
+<script>
+	const ctx = document.querySelector('#newyear').getContext('2d')
+	ctx.fillStyle = 'yellow'
+	ctx.beginPath()
+	ctx.moveTo(100, 10)
+	ctx.lineTo(40, 180)
+	ctx.lineTo(190, 60)
+	ctx.lineTo(10, 60)
+	ctx.lineTo(160, 180)
+	ctx.fill()
+</script>
+```
+
+### Failed
+
+#### Failed Example 1
+
+This `img` element with an empty (`""`) `alt` is not [purely decorative][].
+
+```html
+<img src="/test-assets/shared/w3c-logo.png" alt="" />
+```
+
+#### Failed Example 2
+
+This `img` element with `aria-hidden` set to `true` is not [purely decorative][].
+
+```html
+<img src="/test-assets/shared/w3c-logo.png" aria-hidden="true" alt="W3C logo" />
+```
+
+#### Failed Example 3
+
+This `img` element with an [explicit semantic role][] of `none` is not [purely decorative][].
+
+```html
+<img src="/test-assets/shared/w3c-logo.png" role="none" alt="W3C logo" />
+```
+
+#### Failed Example 4
+
+This `svg` element which has a [semantic role][] of `graphics-document` and an empty (`""`) [accessible name][] is not [purely decorative][].
+
+```html
+Best W3C logo:<br />
+<svg viewBox="0 0 512 512">
+	<path
+		d="M108.4 0h23v22.8h21.2V0h23v69h-23V46h-21v23h-23.2M206 23h-20.3V0h63.7v23H229v46h-23M259.5 0h24.1l14.8 24.3L313.2 0h24.1v69h-23V34.8l-16.1 24.8l-16.1-24.8v34.2h-22.6M348.7 0h23v46.2h32.6V69h-55.6"
+	/>
+	<path fill="#e44d26" d="M107.6 471l-33-370.4h362.8l-33 370.2L255.7 512" />
+	<path fill="#f16529" d="M256 480.5V131H404.3L376 447" />
+	<path
+		fill="#ebebeb"
+		d="M142 176.3h114v45.4h-64.2l4.2 46.5h60v45.3H154.4M156.4 336.3H202l3.2 36.3 50.8 13.6v47.4l-93.2-26"
+	/>
+	<path fill="#fff" d="M369.6 176.3H255.8v45.4h109.6M361.3 268.2H255.8v45.4h56l-5.3 59-50.7 13.6v47.2l93-25.8" />
+</svg>
+```
+
+#### Failed Example 5
+
+This `canvas` element which has no [semantic role][] and an empty (`""`) [accessible name][] is not [purely decorative][].
+
+```html
+<canvas id="w3c" width="200" height="60"></canvas>
+<script>
+	const ctx = document.querySelector('#w3c').getContext('2d')
+	ctx.font = '30px Arial'
+	ctx.fillText('ACT Rules!', 20, 40)
+</script>
+```
+
+### Inapplicable
```

Comment:
Missing examples for the **ignored svg**, **ignored canvas** and **unloaded image** exceptions.

Reply:
Resolved with inapplicable examples 5, 7 and 9

Reply:
Inapplicable examples 5 and 7 are missing description.
And there doesn't seem to be any inapplicable example 9.

Reply:
Darnit, my bad :( forgot to push.

Reply:
I fail to see how examples 5 and 7 illustrate the **ignored svg** and **ignored canvas** exceptions. This may have to do with misunderstanding around the applicability…

* Example 5 is an `svg` which is included in the accessibility tree (with `aria-label`) and thus inapplicable from the start. Additionally, it has an explicit role of `img` while the **ignored svg** exception is about semantic role of `graphics-document`.
* Example 7 is a `canvas` which is included in the accessibility tree (with `aria-label`) and thus inapplicable from the start. Additionally, it has an explicit role while the **ignored canvas** exception is about `canvas` without semantic role.

Reply:
I think it's because of the applicability. Please give it another read.

Reply:
Yes, that's good 👍 
(and I also understand why Inapplicable Examples 5 and 7 were not here from the start, they're kinda not completely needed…)

Reply:
I agree with @Jym77 about examples 5 and 7. Please take another look at exceptions and examples.

Reply:
The first post of this conversation was due to the Applicability.

With the current Applicability, I consider that this conversation has been solved. If @adilsofficial has more to say on the topic, I'll let him handle it 😃